fbchat/_client.py

```diff
@@ -976,6 +976,42 @@ def getUserActiveStatus(self, user_id):
         """
         return self._buddylist.get(str(user_id))
 
+    def _fetchImages(self, thread_id):
+        data = {"id": thread_id, "first": 12}
+        j = self.graphql_request(_graphql.from_query_id("515216185516880", data))
+        to_continue = True
+        while to_continue:
+            page_info = j[thread_id]["message_shared_media"]["page_info"]
+            end_cursor = page_info.get("end_cursor")
```

Comment:
You should only fetch the `end_cursor` when it's needed (aka. further down)

Reply:
Fixed on function merge.